lib/spack/spack/package.py

```diff
@@ -2607,15 +2620,39 @@ def _run_default_install_time_test_callbacks(self):
         if self.install_time_test_callbacks is None:
             return
 
-        for name in self.install_time_test_callbacks:
-            try:
-                fn = getattr(self, name)
-            except AttributeError:
-                msg = 'RUN-TESTS: method not implemented [{0}]'
-                tty.warn(msg.format(name))
-            else:
-                tty.msg('RUN-TESTS: install-time tests [{0}]'.format(name))
-                fn()
+        # Ensure capture of output for tests that use run_test
+        self.test_failures = []
+        self.test_log_file = fsys.join_path(
+            self.stage.path, _spack_install_test_log)
+        fsys.touch(self.test_log_file)  # Otherwise log_parse complains
```

Comment:
This, along with the `log_output` call and the `if self.test_failures` block, feel like they should be factored into a context manager. Could probably be reused in the normal `spack test run` codepath as well.

Reply:
I'll look into that.  Thanks for the feedback.

Reply:
Done